DeepSpeech.py

```diff
@@ -694,594 +651,999 @@ def stopwatch(start_duration=0):
-    ``data_set`` is the selected data set (train, dev, or test).
-    ``tower_results`` is the result of a call to ``get_tower_results()``.
-    ``saver`` is a ``tf.train.Saver`` which can be used to save the model.
-
-    When training an execution context is of the form
-    ``(graph, data_set, tower_results, saver, apply_gradient_op, merged, writer)``.
-    The first four items are the same as in the above case.
-    ``apply_gradient_op`` is an operator that applies the gradents to the learned parameters.
-    ``merged`` contains all summaries for tensorboard.
-    Finally, ``writer`` is the ``tf.train.SummaryWriter`` used to write summaries for tensorboard.
-    """
-    graph = tf.Graph()
-    with graph.as_default():
-        # Set the global random seed for determinism
-        tf.set_random_seed(random_seed)
-
-        # Get the required data set
-        data_set = read_data_set(set_name)
-
-        # Define bool to indicate if data_set is the training set
-        is_train = set_name == 'train'
-
-        # If training create optimizer
-        optimizer = create_optimizer() if is_train else None
-
-        # Get the data_set specific graph end-points
-        tower_results = get_tower_results(data_set, optimizer=optimizer)
-
-        if is_train:
-            # Average tower gradients
-            avg_tower_gradients = average_gradients(tower_results[1])
-
-            # Add logging of averaged gradients
-            if log_variables:
-                log_grads_and_vars(avg_tower_gradients)
-
-            # Apply gradients to modify the model
-            apply_gradient_op = apply_gradients(optimizer, avg_tower_gradients)
-
-        # Create a saver to checkpoint the model
-        saver = tf.train.Saver(tf.global_variables())
-
-        if is_train:
-            # Prepare tensor board logging
-            merged = tf.summary.merge_all()
-            writer = tf.summary.FileWriter(log_dir, graph)
-            return (graph, data_set, tower_results, saver, apply_gradient_op, merged, writer)
+# String constants for different services of the web handler
+PREFIX_NEXT_INDEX = '/next_index_'
+PREFIX_GET_JOB = '/get_job_'
+
+# Global ID counter for all objects requiring an ID
+id_counter = 0
+
+def new_id():
+    '''Returns a new ID that is unique on process level. Not thread-safe.
+
+    Returns:
+        int. The new ID
+    '''
+    global id_counter
+    id_counter += 1
+    return id_counter
+
+class Sample(object):
+    def __init__(self, src, res, loss, accuracy):
+        '''Represents one item of a WER report.
+
+        Args:
+            src (str): source text
+            res (str): resulting text
+            loss (float): computed loss of this item
+            accuracy (float): computed accuracy of this item
+        '''
+        self.src = src
+        self.res = correction(res)
+        self.loss = loss
+        self.accuracy = accuracy
+        self.wer = wer(self.src, self.res)
```

Comment:
This class is mostly a named tuple with a convenience `__str__` method, so I'm weary of adding these calculations (`wer` and specially `correction`) here. The class is only used once, so I'd move these calls to `calculate_report`.